crates/egui/src/widgets/text_edit/builder.rs

```diff
@@ -491,6 +497,11 @@ impl<'t> TextEdit<'t> {
         let mut default_layouter = move |ui: &Ui, text: &str, wrap_width: f32| {
             let text = mask_if_password(password, text);
             let layout_job = if multiline {
+                let wrap_width = if wrap_mode == TextWrapMode::Extend {
+                    f32::INFINITY
+                } else {
+                    wrap_width
+                };
```

Comment:
This logic is now only run for the `default_layouter`, while custom layouters gets the previous `wrap_width` which ignores `wrap_mode`.